src/hotspot/share/opto/movenode.cpp

```diff
@@ -252,6 +308,27 @@ Node *CMoveINode::Ideal(PhaseGVN *phase, bool can_reshape) {
   return n;
 }
 
+Node* CMoveLNode::Ideal(PhaseGVN *phase, bool can_reshape) {
+  // Try generic ideal first
+  Node* x = CMoveNode::Ideal(phase, can_reshape);
+  if (x != nullptr) {
+    return x;
+  }
+
+  // If we're post loop opts then don't attempt to match the min/max pattern, as this node might have been a
+  // MinL or MaxL that was already expanded during macro expansion.
+  if (phase->C->post_loop_opts_phase()) {
+    return nullptr;
+  }
```

Comment:
Putting the condition here would prevent any future optimization further down not to be executed. I think you should rather put this into the `is_minmax` method. Maybe this condition is now only relevant for `long`, but I think it would not hurt to also have it also for `int`, right?

Reply:
That's a good point, I think this will make the logic cleaner. I don't think it'll hurt it to have it for int either.